### DISCUSSION TITLE
V5-beta - Initializing the state machine context

### DISCUSSION THREAD

**Discussion by Simonl9l** [statelyai/xstate#4280] [General]
Per this https://stately.ai/docs/xstate-v5/input#invoking-actors-with-input

and this example:
```ts
import { createActor, createMachine } from 'xstate';

const feedbackMachine = createMachine({
  invoke: {
    src: 'liveFeedback',
    input: {
      domain: 'stately.ai',
    },
  },
}).provide({
  actors: {
    liveFeedback: fromPromise(({ input }) => {
      return fetch(`https://${input.domain}/feedback`).then((res) =>
        res.json(),
      );
    }),
  },
});
```

To confirm that if the state machine has a context defined, the return from `liveFeedback` will need to fully initialize that context?

Or is an assign mixed in here...? with an `onDone` and or `onReject`?

My goal is to have a function (as a service so it awaits than an action), be executed on machine startup to unitize the context. 

My specific use case is that I need to check the `window.location.href` and that is then evaluated as part of an `always` initial state, based on conditions to then have the machine transition to other states based on the href.

All hints and help help appreciated!

**Comment by davidkpiano** [comment]
In this example, it seems like you can add an `invoke.onDone` transition with the `event.output` from the promise, and use that to transition to your desired state.

Would that work for your use-case?
